### DISCUSSION TITLE
🐞 `dagger up --ports` doesn't bind host address when multiple `WithExposedPort` are present in a container

### DISCUSSION THREAD

**Discussion by grouville** [dagger/dagger#12365] [Help]
### What is the issue?

It seems that host ports are not binded properly to the host when multiple `WithExposedPort` are present (and at least one port is binded to the host with `--port`)

### Dagger version

dagger v0.9.10

## Context
Given this example module: 

```go
func (m *MyMod) Package(
) *Service {
	return dag.Container().
		From("localstack/localstack:latest").
		WithExposedPort(4566).
		AsService()
}
```

I can run this command: `dagger call package up --ports 4566:4566`. Querying it with `aws --endpoint-url=http://localhost:4566` works

Now, if I try to run the same container with conflicting ports, whilst the dagger module is running, it fails:
```
docker run -d  -p 4566:4566 -p 4571:4571 localstack/localstack:latest        
43dc09257d16daef95d908ffe11c3108b5f182685e88ef2e5214f97618e52b7c
docker: Error response from daemon: Ports are not available: exposing port TCP 0.0.0.0:4566 -> 0.0.0.0:0: listen tcp 0.0.0.0:4566: bind: address already in use.
```

## Repro
Now, let's add another `withExposedPort`:
```go
func (m *MyMod) Package(
) *Service {
	return dag.Container().
		From("localstack/localstack:latest").
		WithExposedPort(4566).
		WithExposedPort(4571).
		AsService()
}
```

Run the same `dagger up` command as above (with the same host ports, or you can also add 4571 too, it doesn't change the output of next steps):
1. Querying fails: `Could not connect to the endpoint URL: "http://localhost:4566/"`
2. I am able to run the container with conflicting binding ports:
```shell
docker run -d  -p 4566:4566 -p 4571:4571 localstack/localstack:latest        
86f5b1c8e5bd2f63778e00edb244c964300ea350c2aff17f98df353959b04bcd
```

**Comment by vito** [comment]
If I tell port 4571 to skip the health-check it works. Looks like the service isn't binding on that port, and we're missing feedback for that.

```go
package main

type MyMod struct{}

func (m *MyMod) Package() *Service {
	return dag.Container().
		From("localstack/localstack:latest").
		WithExposedPort(4566).
		WithExposedPort(4571, ContainerWithExposedPortOpts{
			ExperimentalSkipHealthcheck: true,
		}).
		AsService()
}
```
